activerecord/lib/active_record/connection_adapters/column.rb

```diff
@@ -26,6 +28,17 @@ def initialize(name, default, sql_type_metadata = nil, null = true, table_name =
         @comment = comment
       end
 
+      def deduplicated
+        @name = -name
+        @table_name = -table_name if table_name
```

Comment:
There are some test which instantiate a column without a `table_name`, I haven't been able to figure out if that's a realistic scenario or not. It seem to be related to dynamic columns when using `AR#select()`.

This makes me wonder if there's a risk of memory leak, some applications could possibly generate very dynamic queries and create new columns over and over.